### PR TITLE
added description about assertions, matchers, custom assertions

### PR DESCRIPTION
**Pull request by IlGrecu**
Added info about assertions/matchers:

- Assertions. General overview
- Hard assertions - Junit
- Matchers - Hamcrest
- Soft assertions - assertj
- Custom assertions

### CODE REVIEW

File: TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
+# Assertions. Matchers. Custom assertions
+
+Author: Ilie Grecu
+---
+
+## Contents
+1. [Assertions. General overview](#assertions-general-overview)
+2. [Hard assertions - Junit](#hard-assertions-junit)
```
Comment (endalexei) [inline TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md:8]: please make sure links work as expected

File: TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+## 1. Assertions. General overview
+
+An assertion is a statement in the Java programming language that enables to test assumptions about a specific functionality 
+of an application. When we develop a test case, we have one expected output and one actual output. Assertion helps us 
+compare these outputs and evaluate whether a given test case is passed or failed.
+Each assertion contains a boolean expression that is believed to be true when the assertion executes. If it is not true, 
+the system will throw an error.
+
+![img.png](resources/assertion representation.png)
```
Comment (endalexei) [inline TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md:21]: this image wont open for me

File: TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+In the first 2 scenarios price for 2 different books is wrong, and in the third one - language for a book is wrong.
+
+In the soft assert case, all 3 scenarios have been executed. All 3 failed and the reason of failure is displayed for each one.
+
+![img.png](resources/soft assert failed scenarios.png)
+
+In the multiple hard assert case only the first one has been executed and it failed. The last 2 haven't been executed 
+and their state is unknown. 
+
+![img.png](resources/multiple hard assert failed scenarios.png)
```
Comment (endalexei) [inline TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md:351]: this image is not loaded for me

File: TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+        });
+    }
+```
+
+Following will be done a comparison on execution result for soft assert and hard assert with 3 assertions and all of them failed. 
+In the first 2 scenarios price for 2 different books is wrong, and in the third one - language for a book is wrong.
+
+In the soft assert case, all 3 scenarios have been executed. All 3 failed and the reason of failure is displayed for each one.
+
+![img.png](resources/soft assert failed scenarios.png)
```
Comment (endalexei) [inline TL/21_assertions_matchers_custom_assertions/Assertions_Matchers_CustomAssertions_README.md:346]: please check that images are seen

**Review by endalexei — CHANGES_REQUESTED** [review]
Nice job, most of the projects require I commit per PR, would you like to try and swash this 2 commits you have ?

**Comment by rudenco** [comment]
Liked the clarity and diversity of explanations + the diversity of libraries used besides JUnit. 
Good examples with Streams and Lamdbas usage in tests.

**Review by bodacd — APPROVED** [review]

**Review by endalexei — APPROVED** [review]

**Pull request merged**